src/qinfer/domains.py

```diff
@@ -0,0 +1,531 @@
+
+    @property
+    def values(self):
+        """
+        Returns an `np.array` of type `self.dtype` containing 
+        some values from the domain.
+        For domains where ``is_finite`` is ``True``, all elements 
+        of the domain will be yielded exactly once.
+
+        :rtype: `np.ndarray`
+        """
+        if self.max is None or self.min is None:
+            return self.example_point
+        else:
+            return np.arange(self.min, self.max + 1, dtype = self.dtype)
+
+    ## METHODS ##
+
+    def in_domain(self, points):
+        """
+        Returns ``True`` if all of the given points are in the domain, 
+        ``False`` otherwise.
+
+        :param np.ndarray points: An `np.ndarray` of type `self.dtype`.
+
+        :rtype: `bool`
+        """
+        return np.all(np.mod(points,1) == 0) and np.all(points >= self._min) and np.all(points <= self._max)
+    
+
+class MultinomialDomain(Domain):
+    """
+    A domain specifying k-tuples of non-negative integers which 
+    sum to a specific value.
+
+    :param int n_meas: The sum of any tuple in the domain.
+    :param int n_elements: The number of elements in a tuple. 
+    """
+
+    def __init__(self, n_meas, n_elements=2):
+        self._n_elements = n_elements
+        self._n_meas = n_meas
+
+    ## PROPERTIES ##
+
+    @property
+    def n_meas(self):
+        """
+        Returns the sum of any tuple in the domain.
+
+        :rtype: `int`
+        """
+        return self._n_meas
+    @property
+    def n_elements(self):
+        """
+        Returns the number of elements of a tuple in the domain.
+
+        :rtype: `int`
+        """
+        return self._n_elements
+    
+
+    @property
+    def is_continuous(self):
+        """
+        Whether or not the domain has an uncountable number of values.
+
+        :type: `bool`
+        """
+        return False
+
+    @property
+    def is_finite(self):
+        """
+        Whether or not the domain contains a finite number of points.
+
+        :type: `bool`
+        """
+        return True
+
+    @property
+    def dtype(self):
+        """
+        The numpy dtype of a single element of the domain.
+
+        :type: `np.dtype`
+        """
+        return np.dtype([('k', np.int, self.n_elements)])
+
+    @property
+    def n_members(self):
+        """
+        Returns the number of members in the domain if it 
+        `is_finite`, otherwise, returns `None`.
+
+        :type: ``int``
+        """
+        return int(binom(self.n_meas + self.n_elements -1, self.n_elements - 1))
+
+    @property
+    def example_point(self):
+        """
+        Returns any single point guaranteed to be in the domain, but 
+        no other guarantees; useful for testing purposes. 
+        This is given as a size 1 ``np.array`` of type ``dtype``.
+
+        :type: ``np.ndarray``
+        """
+        return np.array([([self.n_meas] + list(0 for i in xrange(self.n_elements - 1)),)], dtype=self.dtype)
```

Comment:
Ah, yes. Not sure why I had that there at all for a constant list. And ``n_elements`` refers to the length of the multinomial tuple, and should never be very big.